lib/loomio/loomio_bot.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+# Dreamy ~ the dumb dreams loomio bot
+# imported and used from within the initializer: config/initializers/loomio_bot.rb
+
+# TODO
+# Don't busy loop
+require ::File.expand_path('../../../config/environment', __FILE__)
+require_relative 'processors'
+require_relative 'loomio_handler'
+
+@processors_hash = {
+  'camp_created': Processors.method(:process_camp_created),
+  'other_entry_type': Processors.method(:process_other_entry_type)
+}.stringify_keys
+@predefined_entry_types = @processors_hash.keys
+
+# run through all log_entry_types and run their processors
+def process(log_entry, loomio)
+  puts("Processing %s" % log_entry.description)
+
+  @predefined_entry_types.each do |predefined_entry_type|
+    if log_entry.entry_type == predefined_entry_type
+      @processors_hash[predefined_entry_type].(log_entry, loomio)
+      return
+    end
+
+  puts("Event I don't care about: %s" % log_entry.entry_type)
+  # log_entry.loomio_consumed = true
+
+  puts("Done processing %s" % log_entry.description)
+  end
+end
+
+# used from within the initializer: config/initializers/loomio_bot.rb
+def run_bot
+  if !ENV['LOOMIO_BOT_EMAIL'] || !ENV['LOOMIO_BOT_PASSWORD']
+    puts("LOOMIO_BOT_EMAIL or LOOMIO_BOT_PASSWORD not set. Exiting the loomio bot.")
+    return
+  end
+
+  # initialize the handler here so that the authentication doesn't have to be repeated
+  loomio = LoomioHandler.new(username=ENV['LOOMIO_BOT_EMAIL'], password=ENV['LOOMIO_BOT_PASSWORD'])
+
+  # start a new thread and run an infinite loop inside it
+  Thread.new do
```

Comment:
Also, rather than checking every 5 seconds for new content, I think a more efficient way to do this (how often are folks going to be making camps?) is to create a background job once the camp is created, using ActiveJob
https://edgeguides.rubyonrails.org/active_job_basics.html

(Which was can totally use now that we're on Rails 5!)

The idea would be to queue up a job each time someone creates a camp, and let the app process them in the background as it has time. There's a bunch of different adapters you can use for ActiveJob; sidekick, resque, delayed_job, etc... I'd pick one that looks easy and run with it.

Reply:
The general code for that might look like

```
if create_camp
  log_entry = audit_log(...)
  LoomioThreadJob.perform_later(log_entry.id)
end
```
```
# app/jobs/loomio_thread_job.rb
class LoomioThreadJob < ApplicationJob
  def perform(log_entry_id)
    LogEntry.find(log_entry_id).process!
  end
end
```
(NB that this would probably also remove the need for a loomio_consumed flag on the audit log)